src/org/jitsi/impl/neomedia/rtp/translator/RTCPFeedbackMessageSender.java

```diff
@@ -102,87 +113,291 @@ private long getSenderSSRC()
 
     /**
      * Sends a Full Intra Request (FIR) Command to a media sender/source with a
-     * specific synchronization source identifier (SSRC).
-     * <p>
-     * <b>Warning</b>: Due to (current) limitations in
-     * <tt>RTPTranslatorImpl</tt> and/or <tt>StreamRTPManager</tt>, a FIR
-     * command with the specified <tt>mediaSenderSSRC</tt> is sent to all
-     * <tt>MediaStream</tt>s.
-     * </p>
+     * specific synchronization source identifier (SSRC) through a specific
+     * <tt>MediaStream</tt>.
      *
      * @param mediaSenderSSRC the SSRC of the media sender/source
      * @return <tt>true</tt> if a FIR command was sent; otherwise,
      * <tt>false</tt>
      */
     public boolean sendFIR(int mediaSenderSSRC)
     {
-        boolean sentFIR = false;
-        /*
-         * XXX Currently, the method effectively broadcasts a FIR i.e. sends it
-         * to all streams connected to rtpTranslator because MediaStream's
-         * method getRemoteSourceIds returns an empty list (possibly due to
-         * RED).
-         */
-        for (StreamRTPManager streamRTPManager
-                : rtpTranslator.getStreamRTPManagers())
+        boolean registerRecurringProcessible = false;
+        synchronized (kfRequesters)
```

Comment:
Hi @bickelj, yes, at the cost of having to create a new `KeyframeRequester` every time we attempt to send an FIR. But we only need a limited number of `KeyframeRequester`s per conference (because they're reused), so creating new ones over and over again seems like a waste of CPU cycles.